File: 463-island-perimeter/463-island-perimeter.cpp
class Solution {
public:
    int islandPerimeter(vector<vector<int>>& grid) {
        int perimeter = 0;
        int row = grid.size();
        int col = grid[0].size();
        int nb_neighbors;
        int arr[] = {-1, 1};
        int k;
        // perimeter is equal to 4 - number of neighbors (shared edges)
        for (int i=0; i<row; i++) {
            for (int j=0; j<col; j++) {
                if (grid[i][j]) {
                    nb_neighbors = 0;
                    for (int l=0; l<2; l++) {
                        k = arr[l];  // +/- 1
                        if ((0 <= (i+k)) and ((i+k) < row) and grid[i+k][j]) {
                            nb_neighbors++;
                        }
                        if ((0 <= (j+k)) and ((j+k) < col) and grid[i][j+k]) {
                            nb_neighbors++;
                        }
                    }
                    perimeter = perimeter + (4 - nb_neighbors);
                }
            }
        }
        return perimeter;
    }
};

File: 463-island-perimeter/463-island-perimeter.py
class Solution:
    def islandPerimeter(self, grid: List[List[int]]) -> int:
        perimeter = 0
        row = len(grid)
        col = len(grid[0])
        # perimeter is equal to 4 - number of neighbors (shared edges)
        for i in range(row):
            for j in range(col):
                if grid[i][j]:
                    nb_neighbors = 0
                    for k in (-1, 1):
                        if (0 <= (i+k) < row) and grid[i+k][j]:
                            nb_neighbors += 1
                        if (0 <= (j+k) < col) and grid[i][j+k]:
                            nb_neighbors += 1
                    perimeter += (4 - nb_neighbors)
        return perimeter